clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1245,6 +1248,9 @@ public void handle(OffsetFetchResponse response, RequestFuture<Map<TopicPartitio
 
             if (unauthorizedTopics != null) {
                 future.raise(new TopicAuthorizationException(unauthorizedTopics));
+            } else if (!pendingTxnOffsetTopicPartitions.isEmpty()) {
```

Comment:
We can save this for a follow-up, but it doesn't seem too difficult to allow the consumer to seek to the offsets that were successfully fetched. I think we just need to change the type to something like `Map<TopicPartition, Either<Errors, OffsetAndMetadata>>`.

Reply:
Note that today for the public `#committed / #endOffsets` call if we do not get all the committed offsets after timeout we throw the exception -- i.e. we do not return partial results. So we may return no results and throw timeout but internally silently reset the position for some of the partitions.